superset/charts/commands/delete.py

```diff
@@ -44,6 +45,7 @@ def __init__(self, user: User, model_id: int):
     def run(self) -> Model:
         self.validate()
         try:
+            Dashboard.clear_cache_for_slice(slice_id=self._model_id)
```

Comment:
Can't use `before_delete` event for `Slice` because the association record in the relationship table `datasource_slice` is deleted before the `Slice` object is deleted.